subsys/bluetooth/host/cs.c

```diff
@@ -344,6 +351,74 @@ int bt_le_cs_start_test(const struct bt_le_cs_test_param *params)
 	return bt_hci_cmd_send_sync(BT_HCI_OP_LE_CS_TEST, buf, NULL);
 }
 
+void bt_hci_le_cs_subevent_result(struct net_buf *buf)
+{
+	struct bt_conn *conn = NULL;
+	struct bt_hci_evt_le_cs_subevent_result *evt;
+	struct bt_conn_le_cs_subevent_result result;
+	struct net_buf_simple step_data_buf;
+
+	if (buf->len < sizeof(*evt)) {
+		LOG_ERR("Unexpected end of buffer");
+		return;
+	}
+
+	evt = net_buf_pull_mem(buf, sizeof(*evt));
+
+	if (sys_le16_to_cpu(evt->conn_handle) == BT_HCI_LE_CS_TEST_CONN_HANDLE) {
+		if (!cs_test_callbacks.le_cs_test_subevent_data_available) {
+			LOG_WRN("No callback registered. Discarded subevent results from CS Test.");
+			return;
+		}
+	} else {
+		conn = bt_conn_lookup_handle(sys_le16_to_cpu(evt->conn_handle), BT_CONN_TYPE_LE);
+		if (!conn) {
+			LOG_ERR("Unknown connection handle when processing subevent results");
+			return;
+		}
+	}
+
+	if (evt->subevent_done_status == BT_HCI_LE_CS_SUBEVENT_DONE_STATUS_PARTIAL) {
+		LOG_WRN("Discarded incomplete CS subevent results.");
```

Comment:
Do we need a `bt_conn_unref(conn);` here? I guess this is basically todo anyway so maybe not necessary

Reply:
We might as well. Originally I think I solved this by making this check before the conn handle lookup. So I'm thinking I'll just move it back up

Reply:
I'd still prefer leaving this here and adding `bt_conn_unref`. The reason is that the logic will have a branch here to handle partial results. It doesn't make sense to make handle that branch without making the important checks above first.

Reply:
I was guessing we would end up removing this branch completely and reworking this function later. For now I think we can just keep it simple without if (conn != null) {unref()} and revisit later